File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(input) do
    cond do
      Regex.match?(~r/\?$/, input)
        -> "Sure."
      String.strip(input) == ""
        -> "Fine. Be that way!"
      String.downcase(input) != input and String.upcase(input) == input
        -> "Whoa, chill out!"
      input != nil
        -> "Whatever."
      true -> raise "Your implementation goes here"
    end
  end
end

File: php/bob/bob.php
<?php

class Bob {
    function respondTo(string $input) {
        $input = trim($input);

        $response = 'Whatever.';
        if ($this->isQuestion($input)) {
            if ($this->isYelling($input)) {
                $response = 'Calm down, I know what I\'m doing!';
            }
            else {
                $response = 'Sure.';
            }
        }
        else if (trim($input) === '') {
            $response = 'Fine. Be that way!';
        }
        else if ($this->isYelling($input)) {
            $response = 'Whoa, chill out!';
        }

        return $response;
    }

    protected function isQuestion(string $input) {
        return strpos($input, '?') === (strlen($input) - 1);
    }

    protected function isYelling(string $input) {
        $string = trim(preg_replace('/[^a-zA-Z]/', '', $input));
        if ($string !== '') {
            return $string === strtoupper($string);
        }
        return false;
    }
}